docs/guidelines/migrating-existing-model.md

```diff
@@ -0,0 +1,587 @@
+export default Model;
+```
+
+Using the new implementation patterns strategy, we need to have one builder per data model representation (REST and GraphQL), but here is where we have some extra code comparing it with a brand new data model.
+If we only export two specialized builders, we would not be backwards compatible since current data model consumers are using the unified one. That's why we need to export a third builder which will implement a compatibility version of the new implementation patterns so we can export the same (now legacy) version they are using.
+Example:
+
+```ts
+import {
+  createCompatibilityBuilder,
+  createSpecializedBuilder,
+  TModelFieldsConfig,
+} from '@commercetools-test-data/core';
+import { restFieldsConfig, graphqlFieldsConfig } from './fields-config';
+import type {
+  TCreateChannelBuilder,
+  TChannelGraphql,
+  TChannelRest,
+} from './types';
+
+export const RestModelBuilder: TCreateChannelBuilder<TChannelRest> = () =>
+  createSpecializedBuilder({
+    name: 'ChannelRestBuilder',
+    type: 'rest',
+    modelFieldsConfig: restFieldsConfig,
+  });
+
+export const GraphqlModelBuilder: TCreateChannelBuilder<TChannelGraphql> = () =>
+  createSpecializedBuilder({
+    name: 'ChannelGraphqlBuilder',
+    type: 'graphql',
+    modelFieldsConfig: graphqlFieldsConfig,
+  });
+
+export const CompatChannelModelBuilder = <
+  TChannelModel extends TChannelRest | TChannelGraphql = TChannelRest,
+>() =>
+  createCompatibilityBuilder<TChannelModel>({
+    name: 'ChannelCompatBuilder',
+    modelFieldsConfig: {
+      rest: restFieldsConfig as TModelFieldsConfig<TChannelModel>,
+      graphql: graphqlFieldsConfig as TModelFieldsConfig<TChannelModel>,
+    },
+  });
+```
+
+### Entry point
+
+This is the file that exports the public API of the package. We need to keep the current API and add the new one.
+
+Here is how it currently looks like:
+
+```ts
+export * as Channel from '.';
+
+export { default as random } from './builder';
+export * as constants from './constants';
+export { default as presets } from './presets';
+export * from './types';
+```
+
+With this approach we're exporting a `Channel` object with the builder attached to the `random` property and also the `presets`. This allows consumers to use it like this:
+
+```ts
+import {
+  Channel,
+  type TChannel,
+  type TChannelRest,
+  type TChannelGraphql,
+} from '@commercetools-test-data/channel';
+
+const channel = Channel.random().build<TChannel>();
+const restChannel = Channel.random().buildRest<TChannelRest>();
+const restChannel = Channel.random().buildGraphql<TChannelGraphql>();
+
+// Presets
+const foodStoreChannel = Channel.presets.foodStore().build<TChannel>();
+const restFoodStoreChannel = Channel.presets
+  .foodStore()
+  .buildRest<TChannelRest>();
+const graphqlFoodStoreChannel = Channel.presets
+  .foodStore()
+  .buildGraphql<TChannelGraphql>();
+```
+
+Again, since we want to be backwards, we need to keep that API and add the new builders.
```

Comment:
very nit: i would add the word "compatible" here
Again, since we want to be backwards **compatible**, we need to keep that API and add the new builders.

Reply:
Updated here: 1e3fa34517ade44a57522c11e531f19e02383302

Thanks!